master/internal/resourcemanagers/priority_test.go

```diff
@@ -0,0 +1,251 @@
+		{id: "group1", priority: &lowerPriority},
+		{id: "group2", priority: &higherPriority},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 1, group: groups[1]},
+		{id: "task4", slotsNeeded: 0, group: groups[1]},
+		{id: "task5", slotsNeeded: 4, group: groups[1]},
+		{id: "task6", slotsNeeded: 0, group: groups[0]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, groupMap, agentMap := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	p := &priorityScheduler{}
+	toAllocate, _ := p.prioritySchedule(taskList, groupMap, agentMap, BestFit)
+
+	expectedToAllocate := []*mockTask{tasks[1], tasks[2], tasks[3], tasks[4], tasks[5]}
+	assertEqualToAllocate(t, toAllocate, expectedToAllocate)
+
+	// Check that agent stat has not changed.
+	for _, agent := range agentMap {
+		assert.Equal(t, agent.numEmptySlots(), 4)
+	}
+}
+
+func TestPrioritySchedulingPreemptionDisabledHigherPriorityBlocksLowerPriority(t *testing.T) {
+	lowerPriority := 50
+	higherPriority := 40
+
+	agents := []*mockAgent{
+		{id: "agent1", slots: 4},
+		{id: "agent2", slots: 4},
+	}
+	groups := []*mockGroup{
+		{id: "group1", priority: &lowerPriority},
+		{id: "group2", priority: &higherPriority},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 12, group: groups[1]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, groupMap, agentMap := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	p := &priorityScheduler{}
+	toAllocate, _ := p.prioritySchedule(taskList, groupMap, agentMap, BestFit)
+
+	expectedToAllocate := []*mockTask{}
+	assertEqualToAllocate(t, toAllocate, expectedToAllocate)
+
+	// Check that agent stat has not changed.
+	for _, agent := range agentMap {
+		assert.Equal(t, agent.numEmptySlots(), 4)
+	}
+}
+
+func TestPrioritySchedulingPreemptionDisabledWithLabels(t *testing.T) {
+	lowerPriority := 50
+	higherPriority := 40
+
+	agents := []*mockAgent{
+		{id: "agent1", slots: 4, label: "label1"},
+		{id: "agent2", slots: 4, label: "label1"},
+	}
+	groups := []*mockGroup{
+		{id: "group1", priority: &lowerPriority},
+		{id: "group2", priority: &higherPriority},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0], label: "label1"},
+		{id: "task2", slotsNeeded: 1, group: groups[0], label: "label1"},
+		{id: "task3", slotsNeeded: 4, group: groups[1], label: "label2"},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, groupMap, agentMap := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	p := &priorityScheduler{}
+	toAllocate, _ := p.prioritySchedule(taskList, groupMap, agentMap, BestFit)
+
+	expectedToAllocate := []*mockTask{tasks[0], tasks[1]}
+	assertEqualToAllocate(t, toAllocate, expectedToAllocate)
+}
+
+func TestPrioritySchedulingPreemptionAaron(t *testing.T) {
```

Comment:
q: What is this test name supported to mean?

Reply:
I was re-running just this test and forgot to change the name back, thanks for catching this